src/Kdyby/Doctrine/Proxy/ProxyAutoloaderListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * This file is part of the Kdyby (http://www.kdyby.org)
+ *
+ * Copyright (c) 2008 Filip Procházka (filip@prochazka.su)
+ *
+ * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
+ */
+
+namespace Kdyby\Doctrine\Proxy;
+
+use Kdyby;
+use Nette;
+
+/**
+ * @author Jáchym Toušek <enumag@gmail.com>
+ */
+class ProxyAutoloaderListener extends Nette\Object implements Kdyby\Events\Subscriber
+{
+
+	/** @var array */
+	private $proxyAutoLoaders;
+
+
+
+	public function __construct(array $proxyAutoLoaders)
+	{
+		$this->proxyAutoLoaders = $proxyAutoLoaders;
```

Comment:
@fprochazka Actually your suggestion is quite tricky. Each instance of ProxyAutoloader only deals with one directory which would mean multiple such services. Other way is to let ProxyAutoloader deal with array of directories but that's BC break and unregister method would lose its meaning (you would no longer be able to unregister just one).